src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs

```diff
@@ -14,99 +14,209 @@
+						// - The lowest boundary
+						// - The sum (smallest if there are multiple negative measurements smaller than the lowest
+						// boundary)
+						// - The average in the bucket (smallest if there are multiple positive measurements
+						// smaller than the lowest boundary)
+						return Math.Min(
+							Math.Min(pointData.ExplicitBounds[i], pointData.DoubleValue),
+							pointData.DoubleValue / pointData.LongValue);
+					}
+					return pointData.ExplicitBounds[i - 1];
 				}
 			}
-			catch (DynatraceMetricException e)
+
+			// there are no counts > 0, so calculating a mean would result in a division by 0. By returning
+			// the sum, we can let the backend decide what to do with the value (with a count of 0)
+			return pointData.DoubleValue;
+		}
+
+		private static double GetMaxFromBoundaries(MetricPoint pointData)
+		{
+			// see getMinFromBoundaries for a very similar method that is annotated.
+			if (pointData.BucketCounts.Length == 1)
 			{
-				logger.LogWarning("Skipping metric with the original name '{MetricName}'. Mapping failed with message: {Message}", metric.MetricName, e.Message);
+				if (pointData.BucketCounts[0] > 0)
+				{
+					return pointData.DoubleValue / pointData.LongValue;
+				}
+				return pointData.DoubleValue;
 			}
 
-			if (dynatraceMetric != null)
+			var lastElemIdx = pointData.BucketCounts.Length - 1;
+			// loop over counts in reverse
+			for (var i = lastElemIdx; i >= 0; i--)
 			{
-				yield return dynatraceMetric;
+				if (pointData.BucketCounts[i] > 0)
+				{
+					if (i == lastElemIdx)
+					{
+						// use the last bound in the bounds array. This can only be the case if there is a count >
+						// 0 in the last bucket (lastBound, Inf), therefore, the bound has to be smaller than the
+						// actual maximum value, which in turn ensures that the sum is larger than the bound we
+						// use as max here.
+						return pointData.ExplicitBounds[i - 1];
+					}
+					// in any bucket except the last, make sure the sum is greater than or equal to the max,
+					// otherwise report the sum.
+					return Math.Min(pointData.ExplicitBounds[i], pointData.DoubleValue);
+				}
 			}
-		}
-	}
 
-	/// <summary>
-	/// Combines metric namespace and key into a single key for use in <see cref="Dynatrace.MetricUtils.Metric">.
-	/// </summary>
-	/// <returns>A metric key in the form {MetricNamespace}.{MetricKey}</returns>
-	private static string CreateMetricKey(Metric metric)
-	{
-		var keyBuilder = new StringBuilder();
+			return pointData.DoubleValue;
+		}
 
-		if (!string.IsNullOrEmpty(metric.MetricNamespace))
+		private static IEnumerable<KeyValuePair<string, string>> GetAttributes(this MetricPoint metricPoint, ILogger logger)
 		{
-			keyBuilder.Append($"{metric.MetricNamespace}.");
+			if (metricPoint.Keys == null)
+			{
+				yield break;
+			}
+
+			for (var i = 0; i < metricPoint.Keys.Length; i++)
+			{
+				var value = metricPoint.Values[i];
+
+				if (!(value is string))
```

Comment:
what is this python code doing here :snake:

Reply:
Nice right? If it was more modern c# it could be even better as `if (value is not string)`. But this will have to do for now.